### PR TITLE
Use quasiquotes for attributes in `plugins` crate.

### LINKED ISSUE #5157: Use quasiquote for attributes in plugin crate
Now that rust-lang/rust#19674 is fixed.

### PR DESCRIPTION
**Pull request by utkarshkukreti**
Fixes #5157.

### CODE REVIEW

File: components/plugins/jstraceable.rs [hunk 1]
```diff
@@ -7,27 +7,20 @@ use syntax::codemap::Span;
...
-    {
-        let mut add_attr = |s| {
-            item2.attrs.push(attr::mk_attr_outer(attr::mk_attr_id(), attr::mk_word_item(InternedString::new(s))));
-        };
-        add_attr("must_root");
-        add_attr("privatize");
-        add_attr("jstraceable");
+    item2.attrs.push(quote_attr!(cx, #![must_root]));
+    item2.attrs.push(quote_attr!(cx, #![privatize]));
+    item2.attrs.push(quote_attr!(cx, #![jstraceable]));
```
Comment (Manishearth) [inline components/plugins/jstraceable.rs]: There should not be any exclamation mark

Comment (Manishearth) [inline components/plugins/jstraceable.rs]: (on any of the attributes -- all are being used as outer attrs)

**Comment by Manishearth** [comment]
Looks good, just need to remove the exclamation marks.

Thanks!

**Comment by utkarshkukreti** [comment]
@Manishearth done! (The tests were passing so I didn't notice.)

**Comment by Manishearth** [comment]
Oh, probably because we're directly inserting them so the inner/outer ness doesn't matter

**Comment by Manishearth** [comment]
Rolled up into #5519 because of CI brokenness

**Comment by Ms2ger** [comment]
Rebased as https://github.com/servo/servo/pull/5523

**Pull request closed without merging**